app/models/framework_response.rb

```diff
@@ -112,4 +103,24 @@ def self.recursive_tree
   def set_responded_value
     self.responded = true
   end
+
+  def update_response_transaction(new_value)
+    retried ||= false
+    update!(value: new_value)
+    rebuild_flags!
+    FrameworkResponse.clear_dependent_values_and_flags!([self])
+
+    # lock the status update to avoid race condition on multiple response patches
+    person_escort_record.with_lock do
+      person_escort_record.update_status!
+    end
+  rescue ActiveRecord::PreparedStatementCacheExpired
+    # retry transaction once if this transaction fails
+    if retried
+      raise
+    else
+      retried = true
```

Comment:
Change to `@retried = true` or `self.retried = true` when moving to an instance variable